src/main/java/seedu/address/model/item/ItemQuantity.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.item;
+
+import seedu.address.model.tag.Tag;
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ItemQuantity {
+    public static final String MESSAGE_CONSTRAINTS = "Quantities must be greater than 0, and smaller than 10,000,000.";
+
+    public final int itemQuantity;
+
+    public ItemQuantity(int itemQuantity) {
+        checkArgument(isValidItemQuantity(itemQuantity), MESSAGE_CONSTRAINTS);
+        this.itemQuantity = itemQuantity;
+    }
+
+    public static boolean isValidItemQuantity(int itemQuantity) {
+        return itemQuantity > 0 && itemQuantity < 10000000;
```

Comment:
Is it possible we convert 0 and 1000000 to private static final fields to prevent magic numbers and ease of changing in future. I feel that 0 should be a valid quantity as well. For example, if I always have potatoes, but I accidentally used all of it, I do not want to delete the item.

Reply:
I'll probably rename this to `isValidInputItemQuantity` or something similar. Will change the magic numbers!